### PR TITLE
Remove pip cache in azure pipelines jobs

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit removes the pip cache in azure pipelines t improve the reliability of the CI jobs. Right now the pip cache has grown to be > 4GB and we're seeing a high rate of the step bogging down while downloading the data. Running with a cache in practice has proven to be quite unreliable and just pulling packages from pypi appears to be significantly faster.

### Details and comments

### CODE REVIEW

**Review by jakelishman — DISMISSED** [review]
Perhaps it's worth changing the root of the cache key to something that depends on the date, so we don't have to make this PR periodically? We could (say) take the Unix timestamp modulo 3 months and stick it in there, so the cache autoclears every x period.

If you're not bothered by that, I'm happy just to approve and merge this as-is.

**Comment by mtreinish** [comment]
That's a good idea, I'll push up a commit to do that in the morning

**Comment by mtreinish** [comment]
Actually looking at the py3.7 run I'm wondering if we should just strip the cache completely, it doesn't seem terribly slow to install without one present. Like the install steps were way faster than pulling the cache has been lately. It does open us up to potential networking issue hitting pypi but considering how "reliable" the internal azure pipelines networking has been I'm not sure how much we'd notice.

File: .azure/tutorials-linux.yml [hunk 1]
```diff
@@ -22,8 +22,8 @@ jobs:
         inputs:
           key: 'pip | "$(Agent.OS)" | "${{ parameters.pythonVersion }}"'
           restoreKeys: |
-            pip | "$(Agent.OS)"
-            pip
+            pip-2 | "$(Agent.OS)"
+            pip-2
```
Comment (jakelishman) [inline .azure/tutorials-linux.yml:26]: Did you want to strip this one too?

**Review by jakelishman — COMMENTED** [review]
Matthew and I talked offline and agreed to ditch the cache, at the very least as a trial measure. A full CI run for us pulls on the order of 2GB from PyPI across all jobs, which is substantial for the amount of CI we run per month, but unlikely to hit any rate limits as far as we know. We would prefer to be running our own bandersnatch mirror if we were self-hosting everything, but since we don't have access to that sort of provisioning right now, that's not really an option.

File: .azure/tutorials-linux.yml (hunk 1 above)
Comment (mtreinish) [reply .azure/tutorials-linux.yml:26]: oops I missed the file, let me drop this too

**Review by jakelishman — APPROVED** [review]

**Pull request merged**